### PR TITLE
550 goodid

### PR DESCRIPTION
**Pull request by L03TJ3**
# Description

This adds the goodid and new claim-flow to gooddapp

Todo's:
- [x] - fix goodid details / onboard flow
   - breaks when leaving segmentation early
- [x] - fix styling
- [x] - test mobile
- [ ] - test 3rd party wallets
- [x] - add faq's for claim / goodid

About # (link your issue here)
#550 

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

# Checklist:
- [ ] PR title matches follow: (Feature|Bug|Chore) Task Name
- [ ] My code follows the style guidelines of this project
- [ ] I have followed all the instructions described in the initial task (check Definitions of Done)
- [ ] I have performed a self-review of my own code
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have added reference to a related issue in the repository
- [ ] I have added a detailed description of the changes proposed in the pull request. I am as descriptive as possible, assisting reviewers as much as possible.
- [ ] I have added screenshots related to my pull request (for frontend tasks)
- [ ] I have pasted a gif showing the feature.
- [ ] @mentions of the person or team responsible for reviewing proposed changes

### CODE REVIEW

File: src/index.tsx [hunk 1]
```diff
@@ -7,9 +7,11 @@ import React, { StrictMode } from 'react'
 import ReactDOM from 'react-dom'
 import { Provider } from 'react-redux'
 import { HashRouter as Router } from 'react-router-dom'
-import { AnalyticsProvider } from '@gooddollar/web3sdk-v2'
+import { AnalyticsProvider, GoodIdContextProvider } from '@gooddollar/web3sdk-v2'
 import { NewsFeedProvider } from '@gooddollar/web3sdk-v2'
 import { PostHogProvider } from 'posthog-react-native'
+import { Provider as PaperProvider } from 'react-native-paper'
```
Comment (sirpy) [inline src/index.tsx:13]: since when do we us react-native-paper?

File: src/index.tsx [hunk 2]
```diff
@@ -7,9 +7,11 @@ import React, { StrictMode } from 'react'
...
+import { GoodIdProvider, GoodUIi18nProvider, NativeBaseProvider, RedirectNoticeProvider } from '@gooddollar/good-design'
```
Comment (sirpy) [inline src/index.tsx:14]: can we consolidate this into a single provider?

File: src/components/BlockNativeOnboard/index.tsx [hunk 3]
```diff
@@ -81,7 +81,7 @@ export const OnboardProviderWrapper = ({ children }) => {
     return (
         <OnboardProvider
             options={connectOptions}
-            wallets={{ custom: [torus] }}
+            // wallets={{ custom: [torus] }}
```
Comment (sirpy) [inline src/components/BlockNativeOnboard/index.tsx:84]: why?

File: src/language/index.tsx [hunk 4]
```diff
@@ -34,12 +35,15 @@ export const LanguageContext = React.createContext<{
...
     const [language, setLanguage] = useState(getInitialLocale)
     const [init, setInit] = useState(true)
+    const { setLanguage: setGoodUILanguage } = useGoodUILanguage()
 
     const _setLanguage = useCallback(
         (language: string): void => {
             const switchLocale = (): void => {
                 AsyncStorage.safeSet('lang', language)
                 setLanguage(language)
+                console.log('set language', language)
```
Comment (sirpy) [inline src/language/index.tsx:45]: remove

File: src/pages/gd/Claim/Claim.tsx [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+            <Dialog />
+            <VStack maxWidth="500" margin="auto" mt={0}>
+                <ClaimProvider
+                    activePoolAddresses={
+                        process.env.REACT_APP_UBIPOOL_ADDRESSES
+                            ? JSON.parse(process.env.REACT_APP_UBIPOOL_ADDRESSES)
+                            : {}
+                    }
+                    explorerEndPoints={goodIdExplorerUrls}
+                    supportedChains={[SupportedChains.CELO, SupportedChains.FUSE]}
```
Comment (sirpy) [inline src/pages/gd/Claim/Claim.tsx:77]: should be defaults in claimprovider

File: src/pages/gd/Claim/Claim.tsx [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+                    onNews={onNews}
+                    withNewsFeed
+                    onConnect={handleConnect}
+                    // onSuccess={onClaimSuccess}
+                    onUpgrade={onUpgrade}
+                    withSignModals
+                >
+                    <NewsFeedProvider
```
Comment (sirpy) [inline src/pages/gd/Claim/Claim.tsx:85]: maybe newsfeedprovider should be already inside claimwizard?

File: vite.config.ts [hunk 7]
```diff
@@ -69,13 +81,19 @@ export default defineConfig(({ command, mode }) => {
...
             dedupe: ['react', 'ethers', 'react-dom', 'native-base'],
         },
         build: {
             commonjsOptions: {
+                extensions: ['.js', '.jsx', '.web.js', '.web.jsx'],
                 transformMixedEsModules: true, //handle deps that use "require" and "module.exports"
+                include: [/node_modules/],
+            },
+            rollupOptions: {
+                plugins: [jsxTransform([/react-native-.*\.jsx?$/])], //for some reason react-native packages are not being transpiled even with esbuild jsx settings
```
Comment (sirpy) [inline vite.config.ts:96]: what changed that this is needed?

**Review by sirpy — CHANGES_REQUESTED** [review]

File: src/index.tsx (hunk 2 above)
Comment (L03TJ3) [reply src/index.tsx:14]: @sirpy do you mean as export from the package itself?
or just a separate import into a providers.tsx on protocolUI side?

File: src/pages/gd/Claim/Claim.tsx (hunk 5 above)
Comment (L03TJ3) [reply src/pages/gd/Claim/Claim.tsx:77]: Why? if any changes, we have to release new packages every time before we update/release the dapp?

File: src/index.tsx (hunk 1 above)
Comment (L03TJ3) [reply src/index.tsx:13]: I cannot remember the specifics, should have written them down.
but the native-base modal with React.fragment was not working properly on android devices

I moved it to paper with portal in this commit: 
https://github.com/GoodDollar/GoodWeb3-Mono/pull/128/commits/d33b05328e3a268e9b2cfb6166afb3fc0cc45886

File: src/components/BlockNativeOnboard/index.tsx (hunk 3 above)
Comment (L03TJ3) [reply src/components/BlockNativeOnboard/index.tsx]: Forgot to look into and fix. done here fixing a dep conflict: https://github.com/GoodDollar/GoodProtocolUI/pull/578/commits/77ec17fc8f7921d3dda575c14b729cfdda25d327

File: vite.config.ts (hunk 7 above)
Comment (L03TJ3) [reply vite.config.ts]: I had resolutions issues with one of the dependencies.

Retested now without this, and still works so not sure what changed but seems unnecessary to keep.
https://github.com/GoodDollar/GoodProtocolUI/pull/578/commits/692359963eda04c88b2f188165dbf8d82fa02128

File: src/pages/gd/Claim/Claim.tsx (hunk 6 above)
Comment (L03TJ3) [reply src/pages/gd/Claim/Claim.tsx:85]: maybe, but don't see immediately what we gain from it for now

File: src/language/index.tsx (hunk 4 above)
Comment (L03TJ3) [reply src/language/index.tsx]: done

File: src/components/BlockNativeOnboard/index.tsx (hunk 3 above)
Comment (sirpy) [reply src/components/BlockNativeOnboard/index.tsx]: wasnt that patch required for the custom wallet icons or something?

File: src/pages/gd/Claim/Claim.tsx (hunk 5 above)
Comment (sirpy) [reply src/pages/gd/Claim/Claim.tsx:79]: defaults means they are not required to specify but can be overridden if required

File: src/pages/gd/Claim/Claim.tsx (hunk 6 above)
Comment (sirpy) [reply src/pages/gd/Claim/Claim.tsx:87]: make things less complicated

File: src/index.tsx (hunk 1 above)
Comment (sirpy) [reply src/index.tsx:13]: for modal to work you need to have the provider at the top. I doubt that native-base modals arent android compatible

Comment (L03TJ3) [reply src/index.tsx:13]: @sirpy so you want me to refactor it back to nativebase modals?

the paper modals work fine across all platforms

File: src/index.tsx (hunk 2 above)
Comment (L03TJ3) [reply src/index.tsx]: done here: https://github.com/GoodDollar/GoodProtocolUI/pull/578/commits/8a9f5215d88322cb669869aef878d721b31fc518

File: src/pages/gd/Claim/Claim.tsx (hunk 5 above)
Comment (L03TJ3) [reply src/pages/gd/Claim/Claim.tsx]: obviously.
done here: https://github.com/GoodDollar/GoodProtocolUI/pull/578/commits/cb5214fcd46f843dd2e70aab6423f9b1ba9bd8af

File: src/pages/gd/Claim/Claim.tsx (hunk 6 above)
Comment (L03TJ3) [reply src/pages/gd/Claim/Claim.tsx]: moved to inside postclaim here: https://github.com/GoodDollar/GoodProtocolUI/pull/578/commits/cb5214fcd46f843dd2e70aab6423f9b1ba9bd8af

File: src/index.tsx (hunk 1 above)
Comment (sirpy) [reply src/index.tsx:12]: ok open an issue with link to the commit

**Review by sirpy — APPROVED** [review]

**Pull request merged**